### PR TITLE
AVRO-831 Refactor lang/py setup and test structure

### PR DESCRIPTION
**Pull request by kojiromike**
### Jira

- [x] My PR addresses [AVRO-831](https://issues.apache.org/jira/browse/AVRO-831)
- [x] It references the ticket in the PR title.
- [x] It adds no dependencies. It removes ant and ivy as dependencies.

### Tests

- [x] My PR restructures the tests. (Currently one test is being skipped. I intend to address it.)

### Commits

- [x] My commits all reference Jira issues in their subject lines.
- [x] My commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)".

### Documentation

- [x] My PR does not add documentation, but follows Python standards more closely.

### CODE REVIEW

**Comment by kojiromike** [comment]
@RyanSkraba if you wouldn't mind giving this a look I'd appreciate it.

**Comment by RyanSkraba** [comment]
Excellent timing: I was just reading through the readthedocs on setuptool very recently.  I imagine that this review will be more useful to me than anything else.

One thing I noted already is that ivy no longer downloads the `avro-tools-xxx-SNAPSHOT.jar` from the apache snapshots repository if `lang/py` is built before `lang/java`.   This should only happen in local dev.

This existing behaviour seems wrong to me (depending on a SNAPSHOT artifact published from master when you're potentially on any other dev branch), so I'm OK with it going away.

All of the source changes related to file movement LGTM -- I'm still going through setup.cfg metadata  and setuptool commands!

File: lang/py/avro/test/test_tether_word_count.py [hunk 1]
```diff
@@ -63,6 +64,23 @@ def _version():
...
+  if platform.system() == "Darwin":
+    try:
+      output = subprocess.check_output("/usr/libexec/java_home", stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+      output = e.output
+    return ("No Java runtime present" not in output)
+  return bool(distutils.spawn.find_executable("java"))
+
+
+@unittest.skipUnless(_has_java(), "No Java runtime present")
```
Comment (RyanSkraba) [inline lang/py/avro/test/test_tether_word_count.py]: To clarify my comment: you might also want to skip this test if `_JAR_PATH` doesn't exist.  The current behaviour in the build.xml is to use ivy to *ensure* that it exists (by downloading from apache snapshots if necessary).

**Comment by RyanSkraba** [comment]
Before merge, there's some python ant tasks used `./build.sh` and documented in `gettingstartedpython.xml` (I'm also OK with doing a doc cleanup after merging this -- it looks like that page is a bit out of date and needs some general care).

**Comment by kojiromike** [comment]
all checks are passing now.

File: lang/py/.gitignore [hunk 2]
```diff
@@ -3,3 +3,6 @@
 build/
 lib/
 userlogs/
+avro/HandshakeRequest.avsc
+avro/HandshakeResponse.avsc
+avro/VERSION.txt
```
Comment (RyanSkraba) [inline lang/py/.gitignore]: In setup.py, these files are also copied into this package:

avro/interop.avsc
avro/tether/InputProtocol.avpr
avro/tether/OutputProtocol.avpr

Ignore them as well?

**Comment by RyanSkraba** [comment]
Oops -- `./build.sh clean` and CleanCommand doesn't work.

**Comment by kojiromike** [comment]
> Oops -- `./build.sh clean` and CleanCommand doesn't work.

Oh, good catch. I made another commit. This time I elected to extract the clean command from setup.py entirely and do a minor refactor of build.sh. My rationale is in the commit message.

**Comment by RyanSkraba** [comment]
Sounds good to me!  I rechecked all of the `lang/py/build.sh` targets and all is well.

**Pull request merged**